src/app/util/ember-compatibility-functions.cpp

```diff
@@ -272,6 +273,23 @@ CHIP_ERROR ReadSingleClusterData(FabricIndex aAccessingFabricIndex, const Concre
         return SendFailureStatus(aPath, attributeReport, Protocols::InteractionModel::Status::UnsupportedAttribute, nullptr);
     }
 
+    {
+        Access::SubjectDescriptor subjectDescriptor; // TODO: get actual subject descriptor
+        Access::RequestPath requestPath{ .cluster = aPath.mClusterId, .endpoint = aPath.mEndpointId };
+        Access::Privilege requestPrivilege = Access::Privilege::kView; // TODO: get actual request privilege
+        CHIP_ERROR err                     = Access::GetAccessControl().Check(subjectDescriptor, requestPath, requestPrivilege);
+        if (err != CHIP_NO_ERROR)
+        {
+            AttributeReportIB::Builder attributeReport = aAttributeReports.CreateAttributeReport();
+            ReturnErrorOnFailure(aAttributeReports.GetError());
+            TLV::TLVWriter backup;
+            attributeReport.Checkpoint(backup);
```

Comment:
Why?  We're not writing anything into the buffer, so there is nothing to roll back inside SendFailureStatus.  Just pass nullptr for the last arg there.

Reply:
OK, didn't know that was possible.

Reply:
I was confused by the new IM read chunking and was trying to ensure we were returning proper status. I think I got it now, for wildcards I added an additional backup point for the silent discard. And for now I've removed the access control check for attribute overrides, I definitely want to understand the chunking better, and I might want to refactor where the attribute override occurs in the read attribute code, before adding it there in a subsequent PR.